test/ufuzz.js

```diff
@@ -628,20 +639,30 @@ function createObjectLiteral(recurmax, noComma, stmtDepth, canThrow) {
     return obj + "})";
 }
 
-function createNestedBinaryExpr(recurmax, noComma) {
+function createNestedBinaryExpr(recurmax, noComma, stmtDepth, canThrow) {
     recurmax = 3; // note that this generates 2^recurmax expression parts... make sure to cap it
-    return _createSimpleBinaryExpr(recurmax, noComma);
+    return _createSimpleBinaryExpr(recurmax, noComma, stmtDepth, canThrow);
 }
-function _createSimpleBinaryExpr(recurmax, noComma) {
+function _createSimpleBinaryExpr(recurmax, noComma, stmtDepth, canThrow) {
     // intentionally generate more hardcore ops
     if (--recurmax < 0) return createValue();
     var r = rng(30);
-    if (r === 0) return '(c = c + 1, ' + _createSimpleBinaryExpr(recurmax, noComma) + ')';
-    var s = _createSimpleBinaryExpr(recurmax, noComma) + createBinaryOp(noComma) + _createSimpleBinaryExpr(recurmax, noComma);
-    if (r === 1) {
-        // try to get a generated name reachable from current scope. default to just `a`
-        var assignee = VAR_NAMES[INITIAL_NAMES_LEN + rng(VAR_NAMES.length - INITIAL_NAMES_LEN)] || 'a';
-        return '( ' + assignee + createAssignment() + s + ')';
+    if (r == 0) return '(c = c + 1, ' + _createSimpleBinaryExpr(recurmax, noComma, stmtDepth, canThrow) + ')';
+    var s = _createSimpleBinaryExpr(recurmax, noComma, stmtDepth, canThrow)
+        + createBinaryOp(noComma) + _createSimpleBinaryExpr(recurmax, noComma, stmtDepth, canThrow);
+    if (r <= 3) {
+        var assignee = getVarName();
+        var expr;
+        switch (r) {
+          case 1:
+            return '(' + assignee + createAssignment() + s + ')';
+          case 2:
+            expr = '(' + assignee + '[' + createExpression(recurmax, COMMA_OK, stmtDepth, canThrow)
+                + ']' + createAssignment() + s + ')';
+          case 3:
```

Comment:
Could you make this last one `default:`? I see that the `r == 0` case was handled above but that required too much thinking to verify.